app/src/main/java/io/homeassistant/companion/android/settings/sensor/SensorSettingsViewModel.kt

```diff
@@ -43,22 +46,37 @@ class SensorSettingsViewModel @Inject constructor(
         }
     }
 
+    fun updateManagers(sensorManager: SensorManager) {
```

Comment:
Made an attempt to move the data over to the `init` block however I think due to the `sensors` variable being blank I am unable to generate the `map` that we use in the composable function thus resulting in a blank list of sensors.

Reply:
So while I said "include it in the `init` block", due to how the list is structured that means you'd have to create it in the `filterSensorsList()` function as that is when it changes and you have all the required data. The 'old' implementation in Compose without a suspend function works because it is actually observing the `sensors` variable you're talking about :).


```kotlin
private suspend fun filterSensorsList() {
    // setup context, sensormanager list, basicsensor list
    // create the `sensors` variable from sensormanagers, basicsensors, `sensorsList` and query/filter
    // create the [sensormanager, basicsensor] map from sensormanagers, basicsensors, values in `sensors`
}
```

(sorry for the late response)

Reply:
Thank you very much :) it has been moved and working as expected now!